engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateBy.java

```diff
@@ -1198,7 +1199,10 @@ public static Table updateBy(@NotNull final QueryTable source,
 
         final MutableObject<String> timestampColumnName = new MutableObject<>(null);
         // create an initial set of all source columns
-        final Set<String> preservedColumnSet = new LinkedHashSet<>(source.getColumnSourceMap().keySet());
+        final LinkedHashSet<String> preservedColumnSet = source.getDefinition()
```

Comment:
Why not just use the map constructor that takes a collection, with `getColumnNameSet()`? I guess we don't have to be populating the name map...

Reply:
Yeah, this saves an little bit of extra allocation. Happy to take the less-efficient but prettier approach if you prefer. I don't think we need modifiable results often enough to bake it into TableDefinition, but I could be swayed that way too if we like...

Reply:
I'll take the prettier approach for now; I'm usually the one to argue for maintainability over perceived efficiencies.